matchingengine/OrderBook.go

```diff
@@ -135,7 +135,24 @@ func (ob *orderBook) AddAskOrder(ask *models.Ask) {
 	// in case of stoploss order, add it to stoploss queue and return
 	if ask.OrderType == models.StopLoss {
 		l.Debugf("Adding stopLoss with ask_id %d to the queue", ask.Id)
-		ob.askStoploss.Push(ask)
+		stockCopy, _ := models.GetStockCopy(ask.StockId)
```

Comment:
What is the need for generating a copy here? I'm asking because I'm unfamiliar with this

Reply:
Otherwise you'll have to acquire a lock on the `stock` object, read the price and unlock it. Alternatively you can have a `models.GetCurrentPrice(stockID)`, but `GetStockCopy` was already defined to get a 'snapshot' of the stock at a particular time (useful where you just want current view of the stock object, and don't care if anyone updates the stock after you get your copy)